bundles/org.eclipse.jface/src/org/eclipse/jface/resource/ImageDescriptor.java

```diff
@@ -87,6 +91,21 @@ public static ImageDescriptor createFromFile(Class<?> location, String filename)
 		return new FileImageDescriptor(location, filename);
 	}
 
+	/**
+	 * Creates and returns a new image descriptor from a plug-in.
+	 *
+	 * @param location to determine the bundle which contains the file
+	 * @param filePath path to the file relative in the bundle
+	 * @return a new image descriptor
+	 * @since 3.32
+	 */
+	public static ImageDescriptor createFromBundle(Class<?> location, String filePath) {
+		Bundle bundle = FrameworkUtil.getBundle(location);
+		URL url = FileLocator.find(bundle, IPath.fromOSString(filePath),
```

Comment:
This will most likely throw a NullPointerException in case `FrameworkUtil.getBundle()` returns null. I think an explicit null check, which then simply returns `getMissingImageDescriptor()`, would make sense.